File: coding/15th November 2025/Design HashMap.java
//Design HashMap
class MyHashMap 
{
    int[] myHMap;
    public MyHashMap() 
    {
        myHMap=new int[1000001];
    }
    
    public void put(int key, int value) 
    {
        myHMap[key]=value+1;
    }
    
    public int get(int key) 
    {
        return myHMap[key]-1;
    }
    
    public void remove(int key) 
    {
        myHMap[key]=0;
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */

File: Coding in Python/06th February 2026/Design HashMap.py
#Design HashMap
class MyHashMap:
    def __init__(self):
        self.n = 10000
        self.arr = [[] for _ in range(self.n)]
    def put(self, key: int, value: int) -> None:
        idx = key % self.n
        for i, (k, v) in enumerate(self.arr[idx]):
            if k == key:
                self.arr[idx][i] = (key, value)
                return 
        self.arr[idx].append((key, value))
    def get(self, key: int) -> int:
        idx = key % self.n
        for k, v in self.arr[idx]:
            if k == key:
                return v
        return -1
    def remove(self, key: int) -> None:
        idx = key % self.n
        self.arr[idx] = [(k, v) for k, v in self.arr[idx] if k != key]

# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)
